aten/src/ATen/native/mps/operations/Indexing.mm

```diff
@@ -244,6 +246,57 @@ static Tensor nonzero_fallback(const Tensor& self) {
   return at::nonzero(self.to("cpu")).clone().to("mps");
 }
 
+static Tensor& nonzero_out_native_mps(const Tensor& self, Tensor& out_) {
+
+  using namespace mps;
+
+  int64_t nDim = self.dim();
+  MPSStream* stream = getCurrentMPSStream();
+  struct CachedGraph : public MPSCachedGraph {
+    CachedGraph(MPSGraph* graph) : MPSCachedGraph(graph) {}
+    MPSGraphTensor* inputTensor_ = nil;
+    MPSGraphTensor* outputTensor_ = nil;
+  };
```

Comment:
This is a regular unary op, so let's use a predefined structure from OperationUtils.h
```suggestion
  using CachedGraph = MPSUnaryCachedGraph;
```